### PR TITLE
Simplify the docs site navigation

### LINKED ISSUE #24120: Docs: Redesign the UX for discovering references and architectural information
## Applies To

Planning the exact guides to change is part of the scope of the project. We will need to make a plan first.

## Details

As with other docs sites, Teleport's docs organize pages according to a fourfold taxonomy: getting started guides, how-to guides, references, and conceptual/architecture guides. The information architecture of the Teleport docs site is not consistent in how we organize references and architecture guides.

### Current approaches

We have three different approaches to organizing references and architectural docs:

- **Docs-scoped with dedicated sections:** There are dedicated docs sections for architecture guides (`/docs/architecture/` and references (`/docs/reference/`). These sections are scoped across all of Teleport, with individual guides related to specific topics/themes.
- **Feature-specific subsections:** We organized the Login Rules docs into a subsection of "Access Controls", including a reference, how-to guide, and conceptual introduction. These are specific to Login Rules.
- **Section-wide reference/architecture guide:** A few of our top-level docs sections have an architecture guide or reference guide, e.g., the Database Access section. These are scoped to an entire product area rather than a specific feature.

#### Adoption

Some users would rather have a combination of detailed architectural guides and references than follow how-to guides. These users are experienced engineers and can figure out the necessary commands to run or configuration options to provide as long as they understand how a system works (with architectural guides) and how to interact with it (references). 

It might be the engineer's full-time job for several days or weeks to set up Teleport, giving them time to read architectural guides and references in order to become an expert at Teleport.

We should ensure that engineers who prefer to use references/architecture guides can learn everything they need to about Teleport. Every feature needs to have adequate coverage in a reference guide _and_ architecture guide.

#### Scaling the documentation

As we develop new Teleport features, we need a consistent approach to organizing the documentation for these features so:
- docs authors (who are often/usually engineers) can minimize the time they spend writing new docs
- in general, there's a clear place to put new information

### Possible solution

- Spell out an explicit approach in the docs style guide: There are reasons to prefer each of the three approaches we use above, but we should be consistent and explicit about which approach we choose. Ideally, we should only choose a single approach.
- Consider automatically discovering features that aren't covered in architecture and reference guides. We could do this by:
  - Including feature- and guide-type-related keys in the Markdown frontmatter of each docs page, e.g., `type: architecture` or `type:reference`, plus `feature:login-rules` or `feature:device-trust`. We could then use a script to check whether each feature corresponds to an architecture guide, reference, _and_ how-to guide. This will only work if we scope reference and architecture pages to a single feature.
  -  We could also check to see if any keys inside `yaml` code snippets or flags within `code` snippets within how-to guides are missing in references/architecture guides.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #24120

Provide a less overwhelming documentation experience for users by condensing sidebar sections into five, organized loosely around different audiences of the documentation:

- **Get Started:** new Teleport users, as well as users of all levels who want to quickly look up key, frequently changing information, such as the changelog and upcoming release information.
- **Admin Guides:** intermediate users completing a specific task. This includes all how-to guides in the docs except those in "Get Started" and "Enroll Resources".
- **Enroll Resources:** Admin Guides, but for our extensive library of how-to guides for protecting infrastructure resources with Teleport.
- **References:** Advanced users looking for comprehensive reference information and architecture guides. This includes all reference guides in the docs.
- **User Guides:** Guides for end users looking to access resources.

Related changes:
- **Move Helm reference partials**. Use the `docs/pages/includes` directory so we can generate the sidebar from `docs/pages/reference`.
- **Move the Architecture sidebar section** into "References", and move architecture guides across the documentation into "Architecture".
- **Remove "Documentation Overview"** as this is out of date. Now, top-level sidebar sections are more self explanatory.
- **Remove "Choose an Edition"**, since we want to elevate Teleport Enterprise (Cloud), making Teleport Enterprise (Self-Hosted) available for certain use cases.
- **Add an infrastructure as code section** to Admin Guides.

### CODE REVIEW

**Comment by ptgott** [comment]
**Reviewer note:** I'll add redirects once we backport to v16 since links from Google, the Web UI, etc. point to the default version of the docs site.

**Comment by hugoShaka** [comment]
If you move the generated file you must also edit the generation commands in `examples/chart/Makefile`.

Once this is done, you might want to check the generation still work:
```bash
make -C examples/chart check-chart-ref
make -C examples/chart render-chart-ref
```

**Comment by ptgott** [comment]
> If you move the generated chart references you must also edit the generation commands in `examples/chart/Makefile`.
> 
> Once this is done, you might want to check the generation still work:
> 
> ```shell
> make -C examples/chart check-chart-ref
> make -C examples/chart render-chart-ref
> ```

Thanks! Fixed in 208c27c66bc244df4a5664e402ecd1da593fabac

**Review by ravicious — APPROVED** [review]

**Review by xinding33 — APPROVED** [review]

**Comment by ptgott** [comment]
TODO: Remove redirects when backporting to v15 and v14, since these aren't indexed or used by external Teleport sites/the Web UI.

File: docs/config.json [hunk 1]
```diff
@@ -2,16 +2,16 @@
...
       "entries": [
         {
           "title": "Introduction to Teleport",
           "slug": "/"
         },
-        {
-          "title": "Documentation Overview",
-          "slug": "/documentation-overview/"
-        },
+	{
```
Comment (zmb3) [inline docs/config.json:11]: Looks like you've changed spaces to tabs.

Not a big deal, as this file gets overwritten by post-release workflows all the time, but it does introduce superfluous diffs.

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
FYI, this broke the output of `make -C integrations/terraform docs` due to some changed paths, which in turns breaks certain proto updates from landing (for example, #45166).

FYI @ptgott @hugoShaka.

**Comment by hugoShaka** [comment]
Fix: https://github.com/gravitational/teleport/pull/45175

**Pull request merged**